### PR TITLE
DataGrid - The first cell should not have focus after click if column 'allowEditing' is false and edit mode is 'cell' or 'batch' (T657612)

### PR DESCRIPTION
**Pull request by AntonSermyazhko**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.dataGrid/keyboardNavigation.tests.js [hunk 1]
```diff
@@ -4877,6 +4877,32 @@ QUnit.module("Keyboard navigation with real dataController and columnsController
...
+
+        // act
+        this.gridView.render($("#container"));
+        var keyboardNavigationController = this.gridView.component.keyboardNavigationController;
+        var $cell = $(this.rowsView.element().find(".dx-row").eq(0).find("td").eq(0));
+        $cell.trigger(CLICK_EVENT);
+
+        // assert
+        assert.ok(keyboardNavigationController._isHiddenFocus, "hidden focus");
+        assert.ok($cell.hasClass("dx-cell-focus-disabled"), "cell has no .dx-cell-focus-disabled");
```
Comment (vconst) [inline testing/tests/DevExpress.ui.widgets.dataGrid/keyboardNavigation.tests.js]: has no?

File: testing/tests/DevExpress.ui.widgets.dataGrid/keyboardNavigation.tests.js [hunk 2]
```diff
@@ -4877,6 +4877,32 @@ QUnit.module("Keyboard navigation with real dataController and columnsController
         assert.notOk($cell.hasClass("dx-focused"), "cell has .dx-focused");
     });
 
+    QUnit.testInActiveWindow("Cell should not have focus after click if allowediting=false and edit mode == 'cell || batch' (T657612)", function(assert) {
```
Comment (vconst) [inline testing/tests/DevExpress.ui.widgets.dataGrid/keyboardNavigation.tests.js]: allowediting -> allowEditing, Also conditions are very strange and inconsistent.

**Review by vconst — CHANGES_REQUESTED** [review]

File: testing/tests/DevExpress.ui.widgets.dataGrid/keyboardNavigation.tests.js (hunk 1 above)
Comment (AntonSermyazhko) [reply testing/tests/DevExpress.ui.widgets.dataGrid/keyboardNavigation.tests.js]: Corrected

File: testing/tests/DevExpress.ui.widgets.dataGrid/keyboardNavigation.tests.js (hunk 2 above)
Comment (AntonSermyazhko) [reply testing/tests/DevExpress.ui.widgets.dataGrid/keyboardNavigation.tests.js]: Corrected

**Review by vconst — APPROVED** [review]

**Pull request merged**